fleet/api/http/metrics.go

```diff
@@ -18,71 +19,306 @@ type metricsMiddleware struct {
 	svc     fleet.Service
 }
 
-func (m metricsMiddleware) ViewOwnerByChannelIDInternal(ctx context.Context, channelID string) (fleet.Agent, error) {
+func (m metricsMiddleware) ViewOwnerByChannelIDInternal(ctx context.Context, channelID string) (agent fleet.Agent, _ error) {
+	defer func(begin time.Time) {
+		labels := []string{
+			"method", "viewOwnerByChannelIDInternal",
+			"owner_id", agent.MFOwnerID,
+			"service", "fleet",
+			"id", agent.MFThingID,
+		}
+
+		m.counter.With(labels...).Add(1)
+		m.latency.With(labels...).Observe(time.Since(begin).Seconds())
+
+	}(time.Now())
+
 	return m.svc.ViewOwnerByChannelIDInternal(ctx, channelID)
 }
 
 func (m metricsMiddleware) ViewAgentBackend(ctx context.Context, token string, name string) (interface{}, error) {
+	defer func(begin time.Time) {
+		labels := []string{
+			"method", "viewAgentBackend",
+			"owner_id", "",
```

Comment:
If we don't have a value, I believe we should leave it out of this list, I think we don't have to pass an empty string? Also, we _should_ have owner_id here because we have a token. Let's discuss how to pull it out.